src/App.tsx

```
@@ -8,53 +9,183 @@ import { PostsList } from './components/PostsList';
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
 
-              <Loader />
+  const [loaderPost, setLoaderPost] = useState<LoaderState>('initial');
+  const [loaderComment, setLoaderComment] = useState<LoaderState>('initial');
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const [errorMessagePost, setErrorMessagePost] = useState<string | null>(null);
+  const [errorMessageComment, setErrorMessageComment] = useState<string | null>(
+    null,
+  );
+
+  const [isOpenComment, setIsOpenComment] = useState(false);
+
+  function loadUsers() {
+    usersService.getUsers().then(usersAPI => setUsers(usersAPI));
+  }
+
+  function loadPosts(userId: number) {
+    setLoaderPost('loading');
+    client
+      .get<Post[]>(`/posts?userId=${userId}`)
+      .then(postsAPI => {
+        setPosts(postsAPI);
+        setErrorMessagePost(null);
+      })
+      .catch(() => setErrorMessagePost('Something went wrong!'))
+      .finally(() => {
+        setLoaderPost('loaded');
+      });
+  }
+
+  function loadComments(postId: number) {
+    setLoaderComment('loading');
+
+    client
+      .get<Comment[]>(`/comments?postId=${postId}`)
+      .then(commentsAPI => {
+        setComments(commentsAPI);
+        setErrorMessageComment(null);
+      })
+      .catch(() => setErrorMessageComment('Something went wrong'))
+      .finally(() => {
+        setLoaderComment('loaded');
+      });
+  }
+
+  function addComment({ postId, name, email, body }: Omit<Comment, 'id'>) {
+    return client
+      .post<Comment>('/comments', { postId, name, email, body })
+      .then(newComment => {
+        setComments(currentComments => [...currentComments, newComment]);
+      })
+      .catch(() => {})
+      .finally(() => {});
+  }
```

addComment swallows errors: `.catch(() => {})` and an empty `.finally()`. That hides API failures from the caller/UI and prevents showing an error or providing retry behavior (the task mentions optional handling of Add errors so the user can retry). At minimum propagate the error to the caller (return the rejected promise) or set a local error state so the NewCommentForm can show an add error message / allow retry. Also consider using try/catch or returning the raw promise so the form can set/clear its loading state correctly based on success/failure.